pkgs/tools/text/ripgrep/default.nix

```diff
@@ -31,23 +34,23 @@ rustPlatform.buildRustPackage rec {
   buildFeatures = lib.optional withPCRE2 "pcre2";
 
   preFixup = lib.optionalString enableManpages ''
-    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rg --generate man > rg.1
+    ${rg} --generate man > rg.1
     installManPage rg.1
-  '' + ''
+  '' + lib.optionalString enableShellCompletions ''
```

Comment:
i've asked about this idiom about 6 months ago, and there was a definite, but slim, preference toward exposing these types of options. that providing options for the edge-case behaviors is genuinely useful -- on rare occasion / to a small set of users, and it's cheap to do it.

Reply:
I see the following cases here:

* non-cross (i.e. no emulator used): same as before
* cross, but no emulator available: no reason to request man-pages, the build will fail hard.
* cross, but an emulator available:
  * everything works fine as expected -> no reason to not include man-pages[1]
  * something breaks for whatever reason: that would be a packaging bug IMHO. Also, to unbreak things, one can still override the fixup hook and disable the install-check. I don't really think this is that worse than using the enable* options you've added here because in both cases you'd need to read the source code to find out what to do exactly (i.e. these options appear in no manual and are thus almost not discoverable - a common issue).

Can you think of something else where this option might come in handy?

[1] except maybe to save closure size, but that should be solved by multipled outputs and seems out of scope in this PR. Also, nobody seems to bother in the non-cross case.

Reply:
okay, i've removed both options. `enableManpages` was only added 3 days ago as part of #270521 so i'd be extremely surprised if anyone's relying on it yet.